File: C++/100-200/AR_189.cpp
#include <iostream>
#include <vector>
#include <algorithm>



using namespace std;

// Pre: true
/*int embarque(vector<int>& v, int N) {
    int i = 0, j = 0;
    bool fin = false;

    sort(v.begin(), v.end());


    while (!fin) {
        if (v[i] == N) {
            j = i;
            while (v[i] == N && i != v.size() - 1) {
                ++i;
            }
            fin = true;
            if (i == v.size() - 1) {
                v.erase(v.begin() + j, v.end());
            }
            else {
                v.erase(v.begin() + j, v.begin() + i);
            }
            
        }

        
        ++i;
        if (i == v.size() - 1) fin = true;
    } 

    return v.size();
}*/



int embarque(vector<int>& v, int N) {       
    int i = 0;

   /* while (i < v.size()) {
        if (v[i] == N) {
            v.erase(v.begin() + i, v.begin() + i + 1);
        }
        ++i;
    }*/
    v.erase(remove(v.begin(), v.end(), N), v.end());
    

    return v.size();
}
// Pos: r = Sum k: 0 <= k < v.size(): v[k]

/*long long int consulta(const vector<long long int>& v, int N) {
    return v[N-1];
}*/

bool solve() {
    // read case
    int n;
    cin >> n;
    if (n == 0)
        return false;

    vector<int> v(n);
    for (int i = 0; i < n; ++i)
        cin >> v[i];
        
    int nAcciones;
    cin >> nAcciones;
    string accion;
    int numAccion;
    for (int i = 0; i < nAcciones; ++i) {
        cin >> accion;
        cin >> numAccion;
        if (accion == "EMBARQUE") {            
            cout << embarque(v, numAccion) << "\n";
        }
        else if (accion == "CONSULTA" && v.size() > 0) {
            cout << v[numAccion - 1] << "\n";
        }
    }
    cout << "*\n";
    

    return true;
}

int main() {
    while (solve());
    return 0;
}

File: Java/100-200/AR_189.java
import java.util.*;

public class AR_189 {
    // Pre: true
    public static int embarque(List<Integer> v, int N) {
        
        for (int i=0;i<v.size(); ) {
            if (v.get(i)==N) v.remove(i);
            else i++;
        }
        return v.size();
    }

    // Pos: r = Sum k: 0 <= k < v.size(): v[k]
    public static boolean solve(Scanner sc) {
        // Read case
        int n=sc.nextInt();
        if (n==0) return false;

        List<Integer> v=new ArrayList<>();
        for (int i=0;i<n;i++) v.add(sc.nextInt());

        int nAcciones=sc.nextInt();
        for (int i=0;i<nAcciones;i++) {
            String accion=sc.next();
            int numAccion=sc.nextInt();

            if (accion.equals("EMBARQUE")) {
                System.out.println(embarque(v, numAccion));
            } else if (accion.equals("CONSULTA") && v.size()>0) {
                System.out.println(v.get(numAccion - 1));
            }
        }
        System.out.println("*");
        return true;
    }

    public static void main(String[] args) {
        Scanner sc=new Scanner(System.in);
        while (solve(sc));
        
        sc.close();
    }
}
